### PR TITLE
Refactor to prevent deprecation warnings.

### PR DESCRIPTION
**Pull request by timstoop**
This fixes some deprecation warning and also no longer uses curl to check the status of Jenkins.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -28,12 +28,13 @@
   service: name=jenkins state=started enabled=yes
 
 - name: Wait for Jenkins to start up before proceeding.
-  shell: "curl -D - --silent http://{{ jenkins_hostname }}:{{ jenkins_http_port }}{{ jenkins_url_prefix }}/cli/"
+  uri: url='http://{{ jenkins_hostname }}:{{ jenkins_http_port }}{{ jenkins_url_prefix }}/cli/' status_code=200
   register: result
-  until: (result.stdout.find("403 Forbidden") != -1) or (result.stdout.find("200 OK") != -1) and (result.stdout.find("Please wait while") == -1)
+  until: "(result | success) and (result.status == 200)"
```
Comment (geerlingguy) [inline tasks/main.yml]: Technically we still need to wait until "Please wait while" is not present in the output; Jenkins responds with a 200 OK status even when it's still booting, so we can't proceed until that message is gone.

File: tasks/main.yml [hunk 2]
```diff
@@ -28,12 +28,13 @@
...
   retries: "{{ jenkins_connection_retries }}"
   delay: "{{ jenkins_connection_delay }}"
   changed_when: false
+  no_log: True
```
Comment (geerlingguy) [inline tasks/main.yml]: Huh... TIL; makes the playbook output much more succinct.

File: tasks/plugins.yml [hunk 3]
```diff
@@ -25,5 +25,5 @@
   command: >
     java -jar {{ jenkins_jar_location }} -s http://{{ jenkins_hostname }}:{{ jenkins_http_port }}{{ jenkins_url_prefix | default('') }}/ install-plugin {{ item }}
     creates=/var/lib/jenkins/plugins/{{ item }}.jpi
-  with_items: jenkins_plugins
+  with_items: '{{ jenkins_plugins }}'
```
Comment (geerlingguy) [inline tasks/plugins.yml]: Does this break Ansible 1.8.x or 1.9.x? If so, we need to bump the role requirements to the version this was first supported...

Comment (timstoop) [inline tasks/plugins.yml]: I honestly don't know, I've only recently started using Ansible features like this. But I think it has always been supported. It's just that the non-quoted form is deprecated now,

**Comment by timstoop** [comment]
Add the check for the string, that should work now.

**Comment by geerlingguy** [comment]
Sorry, I think these changes were actually committed in some other update in the past few months; with all the PRs I've been reviewing, this one completely slipped under my radar, and I missed it, sorry about that! This is fixed in the role now.

**Pull request closed without merging**